tsz-compress/src/v2/mod.rs

```diff
@@ -46,8 +1383,832 @@ pub trait TszCompressV2 {
+        // Case 3, 4, 5
+        let values = vec![-32, 115, -78, 56, 12, -127, 89, 43, -3, 101];
+        let (queue_size, num_emitted_samples, decoded_values) =
+            _can_encode_decode_delta_values_i8(&values, false);
+        assert_eq!(num_emitted_samples, 4);
+        assert_eq!(queue_size, values.len() - num_emitted_samples);
+        assert_eq!(
+            values[..num_emitted_samples],
+            decoded_values[..num_emitted_samples]
+        );
+    }
+
+    #[test]
+    fn can_encode_decode_delta_random_i8() {
+        // Random values in i8 range
+        let mut rng = rand::thread_rng();
+        for _ in 0..=100000 {
+            let mut random_vec = Vec::with_capacity(10);
+            for _i in 0..10 {
+                random_vec.push(rng.gen_range(i8::MIN..=i8::MAX));
+            }
+            let (queue_size, num_emitted_samples, decoded_values) =
+                _can_encode_decode_delta_values_i8(&random_vec, true);
+            assert_eq!(queue_size, &random_vec.len() - num_emitted_samples);
+            assert_eq!(
+                random_vec[..num_emitted_samples],
+                decoded_values[..num_emitted_samples]
+            );
+        }
+    }
+
+    #[test]
+    fn can_encode_decode_delta_i8_flush_sanity() {
+        let values = vec![-31, 11, -106, -75];
+        let (queue_size, num_emitted_samples, decoded_values) =
+            _can_encode_decode_delta_values_i8(&values, true);
+        assert_eq!(num_emitted_samples, 4);
+        assert_eq!(queue_size, values.len() - num_emitted_samples);
+        assert_eq!(
+            values[..num_emitted_samples],
+            decoded_values[..num_emitted_samples]
+        );
+    }
+
+    #[test]
+    fn can_encode_decode_delta_i8_flush_sanity2() {
+        let values = vec![93, -127, -100];
+        let (queue_size, num_emitted_samples, decoded_values) =
+            _can_encode_decode_delta_values_i8(&values, true);
+
+        // assert_eq!(num_emitted_samples, 4);
+        assert_eq!(queue_size, values.len() - num_emitted_samples);
+        assert_eq!(
+            values[..num_emitted_samples],
+            decoded_values[..num_emitted_samples]
+        );
+    }
+
+    #[test]
+    fn can_encode_decode_delta_i8_flush_sanity3() {
+        let values = vec![-55, 72];
+        let (queue_size, num_emitted_samples, decoded_values) =
+            _can_encode_decode_delta_values_i8(&values, true);
+
+        assert_eq!(num_emitted_samples, 2);
+        assert_eq!(queue_size, values.len() - num_emitted_samples);
+        assert_eq!(
+            values[..num_emitted_samples],
+            decoded_values[..num_emitted_samples]
+        );
+    }
+
+    #[test]
+    fn can_encode_decode_delta_i8_flush_random() {
+        for _ in 0..100000 {
+            let mut rng = rand::thread_rng();
+            let mut random_vec: Vec<i8> = Vec::with_capacity(10);
+            // Number of samples in flush conditions
+            let end_range = rng.gen_range(1..10);
```

Comment:
The tests should include 0 elements in a slice up to thousands to cover cases of 0, less than 10, 10, over 10, and many random permutations.
```suggestion
            let end_range = rng.gen_range(0..1000);
```